commit e1f6d5fe81c5a07c89af41a12b554c3c839a7dbc
Author: Steve Dickson <steved@redhat.com>
Date:   Mon Jun 22 12:05:12 2015 -0400

    svc_tli_create: close memory leak
    
    xprt->xp_netid can be allocated by makefd_xprt() in
    svc_fd_create(), so don't blindly overwrite the
    pointer in svc_tli_create()
    
    The API expection is the nc_netid passed in will be
    used for the xp_netid. So the pointer must be
    freed then re-allocated.
    
    Reported-by: Michael Theall <mtheall@us.ibm.com>
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/src/svc_generic.c b/src/svc_generic.c
index f49d776..7aae796 100644
--- a/src/svc_generic.c
+++ b/src/svc_generic.c
@@ -283,6 +283,8 @@ svc_tli_create(fd, nconf, bindaddr, sendsz, recvsz)
 	xprt->xp_type = __rpc_socktype2seman(si.si_socktype);
 
 	if (nconf) {
+		if (xprt->xp_netid != NULL)
+			free(xprt->xp_netid);
 		xprt->xp_netid = strdup(nconf->nc_netid);
 		xprt->xp_tp = strdup(nconf->nc_device);
 	}
